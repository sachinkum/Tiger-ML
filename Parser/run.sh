#!/bin/bash

image=sources.x86-linux

if [[ $(uname) = Darwin ]]; then
  image=sources.x86-darwin
fi

# Explicit call to ml-yacc needed here as there is a bug in CM (Compilation Manager).
ml-yacc tiger.grm &&
  ml-build sources.cm Main.main &&
  head -1 tiger.grm.desc &&
  sml @SMLload=${image} \
      ../testcases/*.tig \
      > run.actual.out &&
  diff run.expected.out run.actual.out &&
  echo "good!"

echo status=$?
