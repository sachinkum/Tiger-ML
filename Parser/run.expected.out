../testcases/aaa-nil.tig:
../testcases/aa-test01.tig:
../testcases/aa-test02.tig:
../testcases/aa-test03.tig:
../testcases/aa-test04.tig:
../testcases/aa-test05.tig:
../testcases/aa-test06.tig:
../testcases/aa-test07.tig:
../testcases/aa-test08.tig:
../testcases/aa-test09.tig:
../testcases/aa-test10.fixed.tig:
../testcases/aa-test10.tig:
../testcases/aa-test11.fixed.tig:
../testcases/aa-test11.tig:
../testcases/aa-test12.fixed.tig:
../testcases/aa-test12.tig:
../testcases/aa-test13.tig:
../testcases/aa-test14.tig:
../testcases/aa-test15.tig:
../testcases/aa-test16.tig:
../testcases/aa-test17.tig:
../testcases/aa-test18.tig:
../testcases/aa-test19.tig:
../testcases/merge.tig:
../testcases/queens.tig:
../testcases/test01.tig:
../testcases/test02.tig:
../testcases/test03.tig:
../testcases/test04.tig:
../testcases/test05.tig:
../testcases/test06.tig:
../testcases/test07.tig:
../testcases/test08.tig:
../testcases/test09.tig:
../testcases/test10.tig:
../testcases/test11.tig:
../testcases/test12.tig:
../testcases/test13.tig:
../testcases/test14.tig:
../testcases/test15.tig:
../testcases/test16.tig:
../testcases/test17.tig:
../testcases/test18.tig:
../testcases/test19.tig:
../testcases/test20.tig:
../testcases/test21.fixed.tig:
../testcases/test21.tig:
../testcases/test22.tig:
../testcases/test23.tig:
../testcases/test24.tig:
../testcases/test25.tig:
../testcases/test26.tig:
../testcases/test27.tig:
../testcases/test28.tig:
../testcases/test29.tig:
../testcases/test30.tig:
../testcases/test31.tig:
../testcases/test32.tig:
../testcases/test33.tig:
../testcases/test34.tig:
../testcases/test35.tig:
../testcases/test36.tig:
../testcases/test37.tig:
../testcases/test38.tig:
../testcases/test39.tig:
../testcases/test40.tig:
../testcases/test41.tig:
../testcases/test42.tig:
../testcases/test43.tig:
../testcases/test44.tig:
../testcases/test45.tig:
../testcases/test46.tig:
../testcases/test47.tig:
../testcases/test48.tig:
../testcases/test49.tig:
../testcases/test49.tig:5.18:syntax error: inserting  PLUS
../testcases/test-array-compare-01.tig:
../testcases/test-function-decl-return-unit.tig:
../testcases/test-fun-dec-01.tig:
../testcases/test-mutually-recursive-function-01.tig:
../testcases/test-mutually-recursive-types-01.tig:
../testcases/test-record-compare-01.tig:
../testcases/test-record-compare-02.tig:
../testcases/test-record-exp-01.tig:
../testcases/test-record-exp-02.tig:
../testcases/test-record-exp-03.tig:
../testcases/test-record-exp-04.tig:
../testcases/test-recursive-function-01.tig:
../testcases/test-recursive-type-01.tig:
../testcases/test-recursive-type-02.tig:
